src/widgets/forms/autocomplete/publicData.ts

```diff
@@ -290,6 +293,13 @@ export async function queryESCODataByName (filter: string, theClass:NamedNode, q
   return ESCOResultToBindings(json)
 }
 
+function fixWikidataJSON (str) {
+  const badness = str.indexOf('SPARQL-QUERY')
+  if (badness < 0) return str
+  debug.warn('  ### Fixing JSON with wikidata error code injection ' + str.slice(badness, badness + 200))
```

Comment:
**question:** what is "badness" why is there a magic number of 200 added? What is it about?

**suggestion:** use named constants instead of magic numbers.

Reply:
"Badness is a flag that the returned entity is not valid JSON but is a horrible concatenation valid JSON and and timeout error message form a different level of the server system This is a massive kludge, because wikidata will return a huge list of bindings and then interrupt it with an error string starting "SPARQL-QUERY' when the server process times out.  The JSON parser which comes by default with the JS world throws a syntax error and so does not return any data, which is ironic because in these cases the number of bindings returned before the timeout is typically very large.  How best to deal with this? Link a wikidata issue and a solid-ui issue?  
The 200 is just an arbitrary length of the error message received passed onto the debug.warn stream .... there is normally a lot more, including a Java stack IIRC.

Reply:
ok, ugly thing... so, what we could do about it is at least leave this helpful explanation you gave here as a comment in the code so that people can find it. Or yes, create issues on wikidata/and or solid-ui and leave a link to the issue as a comment in the code.

Reply:
Re-written with diff variable name.   Included a test using the bad data to make sure it works.

Reply:
Re-written and test added to make sure it works.